### ISSUE TITLE
libshmem_pmi should not be named as such or need symbolic link from libshmem

### ISSUE THREAD

**Issue by jeffhammond** [Sandia-OpenSHMEM/SOS#82]
PRK should not have to know that SOS was build with PMI included...

``` sh
[jrhammon@esgmonster github-official]$ make allshmem
cd SHMEM/Synch_p2p;         make p2p       "DEFAULT_OPT_FLAGS   = -O3"
make[1]: Entering directory `/home/jrhammon/Work/INTEL/PCL/ESG/PRK/github-official/SHMEM/Synch_p2p'
/opt/shmem/sandia/intel/bin/oshcc -I/opt/shmem/sandia/intel/include -O3 -DSHMEM     -I../../include -c p2p.c
/opt/shmem/sandia/intel/bin/oshcc -I/opt/shmem/sandia/intel/include -O3 -DSHMEM     -I../../include -c ../../common/wtime.c
/opt/shmem/sandia/intel/bin/oshcc -I/opt/shmem/sandia/intel/include -O3 -DSHMEM     -I../../include -c ../../common/SHMEM_bail_out.c
/opt/shmem/sandia/intel/bin/oshcc -I/opt/shmem/sandia/intel/include  -o p2p -L/opt/shmem/sandia/intel/lib  -O3 -DSHMEM p2p.o wtime.o SHMEM_bail_out.o  -lshmem
ld: cannot find -lshmem
make[1]: *** [p2p] Error 1
make[1]: Leaving directory `/home/jrhammon/Work/INTEL/PCL/ESG/PRK/github-official/SHMEM/Synch_p2p'
make: *** [allshmem] Error 2
```

``` sh
[jrhammon@esgmonster github-official]$ ll /opt/shmem/sandia/intel/lib
total 2.6M
-rw-r--r-- 1 root root 114K Dec 22 09:10 libshmem_pmi.a
-rwxr-xr-x 1 root root  958 Dec 22 09:10 libshmem_pmi.la
lrwxrwxrwx 1 root root   21 Dec 22 09:10 libshmem_pmi.so -> libshmem_pmi.so.0.0.0
lrwxrwxrwx 1 root root   21 Dec 22 09:10 libshmem_pmi.so.0 -> libshmem_pmi.so.0.0.0
-rwxr-xr-x 1 root root  77K Dec 22 09:10 libshmem_pmi.so.0.0.0
-rw-r--r-- 1 root root 1.6M Dec 22 09:10 libsma.a
-rwxr-xr-x 1 root root  996 Dec 22 09:10 libsma.la
lrwxrwxrwx 1 root root   15 Dec 22 09:10 libsma.so -> libsma.so.0.0.0
lrwxrwxrwx 1 root root   15 Dec 22 09:10 libsma.so.0 -> libsma.so.0.0.0
-rwxr-xr-x 1 root root 821K Dec 22 09:10 libsma.so.0.0.0
```

**Comment by jdinan** [comment]
I agree; you should use oshcc.  This is the suggestion made in Annex B of the OpenSHMEM spec.

The only other fix for this issue would be for SOS to print out the dependencies after the build, so users can build without the wrapper.  PRK should also not expect the SHMEM library to be called libshmem.

The way I build the PRK is ```make CC=oshcc LDLIBS=-lm```.

**Comment by jeffhammond** [comment]
I am building with the wrapper `/opt/shmem/sandia/intel/bin/oshcc`...

**Comment by jdinan** [comment]
Sorry, in that case there was an error in wrapper generation.  Is this also with static linking on Linux?  I think we have only been testing the dynamic linking build.

**Comment by jeffhammond** [comment]
Yeah these issues are both appearing with static linking on Linux. PRK tries to link libshmem but it isn't required with the wrappers. 

Can wrappers use rpath so I don't need to set LD_LIBRARY_PATH? That's the best fix here.

**Comment by jdinan** [comment]
We can look into it.  I think the rpath fix would go into ```oshrun```.

**Comment by jeffhammond** [comment]
rpath is a linked option (https://ftp.gnu.org/old-gnu/Manuals/ld-2.9.1/html_node/ld_3.html).

You could prepend  LD_LIBRARY_PATH appropriately in oshrun instead of rpath...

**Comment by jeffhammond** [comment]
This will be fixed in part by #84.

**Comment by jdinan** [comment]
Fixed in #346
